src/libYARP_sig/include/yarp/sig/PointCloud.h

```diff
@@ -0,0 +1,597 @@
+            data.resize(width * height);
+        }
+
+        /**
+         * @brief Resize the PointCloud.
+         * @param width.
+         * Calling this function the cloud will become NOT organizedS
+         */
+        virtual void resize(size_t width)
+        {
+            header.width = width;
+            header.height = 1;
+            data.resize(width);
+        }
+
+        /**
+         * @brief Get the pointer to the data.
+         * @return the pointer to the data.
+         */
+        virtual const char* getRawData() const override
+        {
+            return data.getMemoryBlock();
+        }
+
+        /**
+         * @brief Get the size of the data + the header in terms of
+         * number of bytes.
+         * @return the size of the data sent through the network
+         */
+        virtual size_t wireSizeBytes() const override
+        {
+            return sizeof(header) + dataSizeBytes();
+        }
+
+        /**
+         * @brief Get the size of the data in terms of
+         * number of bytes.
+         * @return the size of the data
+         */
+        virtual size_t dataSizeBytes() const override
+        {
+            return header.width*header.height*(sizeof(T));
+        }
+
+        /**
+         * @brief
+         * @return the number of points of the PointCloud
+         */
+        virtual size_t size() const override
+        {
+            return data.size();
+        }
+
+        /**
+         * @brief Obtain the point given by the (column, row) coordinates.
+         * Only works on organized clouds (those that have height != 1).
+         * @param u, column coordinate
+         * @param v, row coordinate
+         */
+        inline T& operator()(size_t u, size_t v) {
+            yAssert(isOrganized());
+            if (u > width() || v > height())
+            {
+                return nulldata;
+            }
+            return data[u + v*width()];
+        }
+
+        /**
+         * @brief Obtain the point given by the index.
+         * @param i, index
+         */
+        inline T& operator()(size_t i) {
+            if (i > data.size())
+            {
+                return nulldata;
+            }
+            return data[i];
+        }
+
+        template<class T1>
+        /**
+         * Assignment operator.
+         * @brief Clones the content of another image.
+         * @param alt the image to clone
+         */
+        const PointCloud<T>& operator=(const PointCloud<T1>& alt)
+        {
+            copy(alt);
+            return *this;
+        }
+
+
+        /**
+         * \brief Concatenate a point cloud to the current cloud.
+         *  \param rhs the cloud to add to the current cloud
+         * \return the new cloud as a concatenation of the current cloud and the new given cloud
+         */
+        inline PointCloud<T>&
+        operator+=(const PointCloud<T>& rhs)
+        {
+            //TODO: take the newest timestamp
```

Comment:
If this is an actual TODO, can you open a issue in the YARP issue tracker?

Reply:
I think that I can remove it, in PCL the `operator+=` takes the newest timestamp because the timestamp is in the pointcloud header, but in `YARP`  is handled on another level.